lib/backup_restore/backup_store.rb

```diff
@@ -0,0 +1,76 @@
+module BackupRestore
+  # @abstract
+  class BackupStore
+    class BackupFileExists < RuntimeError; end
+    class StorageError < RuntimeError; end
+
+    # @return [BackupStore]
+    def self.create(opts = {})
+      case SiteSetting.backup_location
+      when 'local'
+        require_dependency "backup_restore/local_backup_store"
+        BackupRestore::LocalBackupStore.new(opts)
+      when 's3'
+        require_dependency "backup_restore/s3_backup_store"
+        BackupRestore::S3BackupStore.new(opts)
+      end
+    end
+
+    # @return [Array<BackupFile>]
+    def files
+      unsorted_files
+        .sort_by { |file| file.last_modified }
+        .reverse
```

Comment:
Hmm sorting and then reversing seems weird. Will `unsorted_files.sort_by { |file| -file.last_modified }`?